src/main/java/com/mraof/minestuck/event/ServerEventHandler.java

```diff
@@ -277,13 +281,70 @@ public static void onPlayerTickEvent(TickEvent.PlayerTickEvent event)
 			if(data.getTitle() != null)
 				data.getTitle().handleAspectEffects((ServerPlayerEntity) event.player);
 		}
+		
+		if(!event.player.isCreative() && event.player.isPotionActive(MSEffects.CREATIVE_SHOCK.get()))
+		{
+			int duration = event.player.getActivePotionEffect(MSEffects.CREATIVE_SHOCK.get()).getDuration();
+			if(duration >= 5)
+				event.player.abilities.allowEdit = false;
+			else
+			{
+				if(!event.player.world.isRemote)
+					event.player.abilities.allowEdit = ((ServerPlayerEntity) event.player).interactionManager.getGameType().hasLimitedInteractions();
+			}
+		}
+	}
+	
+	@SubscribeEvent
+	public static void onEffectRemove(PotionEvent.PotionRemoveEvent event)
+	{
+		//onEffectEnd(event.getEntityLiving(), event.getPotionEffect().getPotion());
 	}
 	
+	@SubscribeEvent
+	public static void onEffectExpire(PotionEvent.PotionExpiryEvent expiryEvent)
+	{
+		//onEffectEnd(expiryEvent.getEntityLiving(), expiryEvent.getPotionEffect().getPotion());
+	}
+	
+	private static void onEffectEnd(LivingEntity entityLiving, Effect effect) //TODO MSPacketHandler.sendToPlayer recieves an invalid message
+	{
+		if(entityLiving instanceof PlayerEntity)
+		{
+			PlayerEntity player = (PlayerEntity) entityLiving;
+			
+			if(player instanceof ServerPlayerEntity && !player.isCreative() && effect == MSEffects.CREATIVE_SHOCK.get())
+			{
+				ServerPlayerEntity serverPlayerEntity = (ServerPlayerEntity) player;
+				player.abilities.allowEdit = !serverPlayerEntity.interactionManager.getGameType().hasLimitedInteractions();
+				
+				StopCreativeShockEffectPacket packet = new StopCreativeShockEffectPacket(serverPlayerEntity.interactionManager.getGameType());
+				MSPacketHandler.sendToPlayer(packet, serverPlayerEntity);
+			}
```

Comment:
Any idea how to make it send a packet it will accept?

Reply:
The packet is considered invalid because it is not registered in `MSPacketHandler`